pkg/api/v1alpha1/condition_consts.go

```diff
@@ -0,0 +1,49 @@
+package v1alpha1
+
+// Conditions and condition Reasons for the Cluster object.
+const (
+	// ReadyCondition reports a summary of other conditions, indicating an overall operational
+	// state of the cluster.
```

Comment:
I think we need to extend this. Like reflect precessily what it means: all cp nodes in the right version, all worker nodes in the right version, all nodes ready, not old nodes